lib/Logger.js

```diff
@@ -14,6 +14,8 @@ module.exports = class Logger {
     this._transports = transports;
     this._threshold = level;
     this._disabled = false;
+    this._promises = new Set();
```

Comment:
Maybe the name should be more description, e.g. `this._pendingWrites`

Reply:
Yes, there could be other lists of promises you might need to store in the future.

Reply:
I've changed to _pendingTransports